### ISSUE TITLE
Allow custom loss functions and optimization in Neural Network Predictor

### ISSUE THREAD

**Issue by danyoungday** [Project-Resilience/sdk#11]
Currently the Neural Network Predictor defaults to AdamW optimizer with L1Loss, however we would like to be able to customize this.
